MiniShopping/MiniShopping/Models/FetchProducts.m

```diff
@@ -0,0 +1,43 @@
+#import "FetchProducts.h"
+#import "ImageDownloader.h"
+
+@implementation FetchProducts
+
+- (void)fetchData {
+    NSString *urlString = @"https://dummyjson.com/products";
+    NSURL *url = [NSURL URLWithString:urlString];
+    
+    NSURLSessionTask *task = [[NSURLSession sharedSession] dataTaskWithURL:url completionHandler:^(NSData * _Nullable data, NSURLResponse * _Nullable response, NSError * _Nullable error) {
+        if (error) {
+            NSLog(@"URL Session Error :  %@", error);
+            return;
+        }
+        if (!data) {
+            NSLog(@"Data Error :  %@", error);
+            return;
+        }
```

Comment:
For network request response handling, we should also check the response status code (200 - OK/success, 400 - bad request, etc) to handle those errors gracefully.